Added Gazebo Simulation for Kuka LBR iisy 3 R760

jacob-kruse opened the pull request:
### Before submitting this PR into master please make sure:
If you added a new robot model:
- [ ] you extended the table of verified data in `README.md` with the new model
- [ ] you extended the CMakeLists.txt of the appropriate moveit configuration package with the new model
- [ ] you added a `test_<robot_model>.launch.py` and after launching the robot was visible in `rviz`
- [ ] you added a `<robot_model>_joint_limits.yaml` file in the `config` directory (to provide moveit support)

If you modified an already existing robot model:
- [x] you checked and optionally updated the table of verified data in `README.md` with the changes
- [x] you have run the `test_<robot_model>.launch.py` and the robot was visible in `rviz`

### Short description of the change
Added a Gazebo Launch and Gazebo URDF with macros for Gazebo Simulation with the Kuka LBR iisy 3 R760

Note: Inertial values in the gazebo_macro were required for Gazebo. These values are just placeholders and not accurate by any means.

Svastits commented:
Thanks for your contribution :)
We do not have a policy yet for gazebo urdf, launch and config files, so we might take some time to consider the most optimal structure for that to minimize code duplication and maximize the user experience.

is the controller_manager node not necessary? the controllers cannot be spawned and therefore I can not see the robot neither in rviz, nor in gazebo

jacob-kruse commented:
The `controller_manager` node is not necessary because Gazebo has its own controller manager. When I run the launch file, the controllers spawn correctly, I see the robot in both RViz and Gazebo, and I am able to motion plan. 

I tried the launch without the controller spawners and did not see the URDF's, so I am assuming there is some issue with the Gazebo controller manager in your situation. 

Make sure you have ros-jazzy-gz-ros2-control installed for the control plugin and all the Gazebo simulation packages installed. The commands below should install these. 

`sudo apt install ros-jazzy-gz*`
`sudo apt install ros-jazzy-ros-gz*`

jacob-kruse commented:
I fixed the formatting issues in my most recent commit. I also implemented the lbr_iisy11_r1300 and lbr_iisy15_r930 for Gazebo simulation. I have tested these models and I am able to motion plan in Gazebo with both additions.

In response to your earlier point about code organization, it might be good to make a folder for Gazebo simulation. This folder could store the launch file, configuration files, and Gazebo macros. The URDF's I added are essentially the same, I only added inertial values and calls to the Gazebo hardware and Gazebo control plugins. At some point the original URDF files can be modified and passed an argument for Gazebo simulation to call the macros.

Groebehavn commented:
Dear @jacob-kruse ,
We added Gazebo support for multiple robot families in a recent [PR](https://github.com/kroshu/kuka_robot_descriptions/pull/82).
_Jazzy_ support is now there, _Humble_ soon follows. You can check out the newly merged content which hopefully gives you support for the iisy.
Nevertheless, thank you again for your contributions!
This PR will then be closed.
